packages/matchbox/src/components/Pagination/Pagination.js

```diff
@@ -48,21 +48,23 @@ function Pagination(props) {
   }
 
   function _createButtons(start, end) {
-    const buttons = [];
-
-    for (let currentIndex = start; currentIndex < end; currentIndex++) {
-      buttons.push({
-        content: `${currentIndex + 1}`,
-        onClick: handlePageChange.bind(this, currentIndex),
-        flat: currentIndex === index ? false : true,
-        size: 'small',
-        color: currentIndex === index ? 'blue' : 'gray',
-        mx: '100',
-        width: currentIndex + 1 < 100 ? '32px' : 'auto',
-      });
-    }
-
-    return buttonsFrom(buttons);
+    return React.useMemo(() => {
+      const buttons = [];
+
+      for (let currentIndex = start; currentIndex < end; currentIndex++) {
+        buttons.push({
+          content: `${currentIndex + 1}`,
+          onClick: handlePageChange.bind(this, currentIndex),
+          flat: currentIndex === index ? false : true,
+          size: 'small',
+          color: currentIndex === index ? 'blue' : 'gray',
+          mx: '100',
+          width: currentIndex + 1 < 100 ? tokens.spacing_600 : 'auto',
+        });
+      }
+
+      return buttonsFrom(buttons);
```

Comment:
Not really part of this ticket but noticed buttons are missing `cursor: pointer` 😁